### PR TITLE
feat: use standard forge directory structure

### PR DESCRIPTION
**Pull request by marktoda**
- feat: standard foundry directory structure
- feat: remove JS stuff

## Related Issue
Which issue does this pull request resolve?

## Description of changes

### CODE REVIEW

**Review by hensha256 — COMMENTED** [review]
Mostly looking great! Just 2 comments :)

File: yarn.lock [hunk 1]
```diff
@@ -2,177 +2,163 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@^7.12.13", "@babel/code-frame@^7.16.0":
```
[the remark this answers is not in the document]
Comment (marktoda) [reply yarn.lock:5]: Oh I tried deleting it but it's needed for the sqrt js comparison tests with ffi. Can prune the deps and eventually move into a fully independent subdir

File: test/PoolManager.gas.spec.ts [hunk 2]
```diff
@@ -1,566 +0,0 @@
-// import { ethers, waffle } from 'hardhat'
-// import { Wallet } from 'ethers'
-// import { PoolManager, PoolSwapTest, PoolDonateTest, PoolModifyPositionTest } from '../typechain'
```
[the remark this answers is not in the document]
Comment (marktoda) [reply test/PoolManager.gas.spec.ts]: oop brought back

**Review by hensha256 — APPROVED** [review]
lgtm 🚀

**Pull request merged**